src/sinks/gcs_common/service.rs

```diff
@@ -39,28 +41,22 @@ impl GcsService {
 
 #[derive(Clone, Debug)]
 pub struct GcsRequest {
+    pub key: String,
     pub body: Bytes,
     pub settings: GcsRequestSettings,
-    pub metadata: GcsMetadata,
-}
-
-#[derive(Clone, Debug)]
-pub struct GcsMetadata {
```

Comment:
Ah, I see you removed this struct. Is that because you dropped its use above, or is the causality the reverse?

Reply:
Factored out the custom type back to being just a tuple, which matches the other implementations of `RequestBuilder`. It was also going to look weird since the request builder was setting `GcsMetadata` as a field on `GcsRequest`.. and then I was going to be setting another field as a `BatchRequestMetadata`.

Just felt weird/repetitive.